server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.blobstore;
+
+import org.apache.lucene.store.IndexInput;
+import org.opensearch.common.blobstore.AsyncMultiStreamBlobContainer;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+import org.opensearch.common.blobstore.transfer.RemoteTransferContainer;
+import org.opensearch.common.blobstore.transfer.stream.OffsetRangeIndexInputStream;
+import org.opensearch.common.lucene.store.ByteArrayIndexInput;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * Standard format that has only content for writes. Read interface does not exist as it not yet required.
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreFormat<T extends ToXContent> {
```

Comment:
why did you need this new class?

Reply:
The existing class `ChecksumBlobStoreFormat` has additional metadata in the file being uploaded to remote store. The additional metadata includes header and footer. These additional data is derived using Lucene's CodecUtil. We do not need the header since we are maintaining version information in the content as the first field. We dont need the footer since the data is sitting in memory and does not require any further checksumming while upload. S3 sdk provides checksumming while download on the client side as well. And moreover, these (header and footer) are derived using lucene's CodecUtil class which does not have a official support for all other languages. Hence came up with this class that does not add any additional metadata to the content being uploaded. Other option was to 1) overload methods in the existing `ChecksumBlobStoreFormat` class but that would defeat the purpose of the name 2) directly upload from the `RemoteIndexPathUploader` class but this would dilute the need for separation of logic from unnecessary details. Considering all this, created this class that can be used for other usecases as well in future.

Reply:
Done.